api/v1alpha1/package_types.go

```diff
@@ -55,6 +55,14 @@ func (pkg *Package) SetAutoUpdatesEnabled(enabled bool) {
 	setAutoUpdatesEnabled(&pkg.ObjectMeta, enabled)
 }
 
+func (pkg *Package) InstalledAsDependency() bool {
+	return false
+}
+
+func (pkg *Package) SetInstalledAsDependency(value bool) {
+	panic("illegal operation: package can not be installed as dependency")
```

Comment:
I don't know, my rationale to do it like this was that it would most likely be a programmer error if this was used and by adding a panic we would probably discover it more easily.